helper/logging/logging_http_transport.go

```diff
@@ -0,0 +1,279 @@
+func NewSubsystemLoggingHTTPTransport(subsystem string, t http.RoundTripper) *loggingHttpTransport {
+	return &loggingHttpTransport{subsystem, t}
+}
+
+const (
+	// FieldHttpOperationType is the field key used by NewLoggingHTTPTransport
+	// and NewSubsystemLoggingHTTPTransport when logging the type of HTTP operation via tflog.
+	FieldHttpOperationType = "tf_http_op_type"
+
+	// OperationHttpRequest is the field value used by NewLoggingHTTPTransport
+	// and NewSubsystemLoggingHTTPTransport when logging an HTTP request via tflog.
+	OperationHttpRequest = "request"
+
+	// OperationHttpResponse is the field value used by NewLoggingHTTPTransport
+	// and NewSubsystemLoggingHTTPTransport when logging an HTTP response via tflog.
+	OperationHttpResponse = "response"
+
+	// FieldHttpRequestMethod is the field key used by NewLoggingHTTPTransport
+	// and NewSubsystemLoggingHTTPTransport when logging an HTTP request method via tflog.
+	FieldHttpRequestMethod = "tf_http_req_method"
+
+	// FieldHttpRequestUri is the field key used by NewLoggingHTTPTransport
+	// and NewSubsystemLoggingHTTPTransport when logging an HTTP request URI via tflog.
+	FieldHttpRequestUri = "tf_http_req_uri"
+
+	// FieldHttpRequestProtoVersion is the field key used by NewLoggingHTTPTransport
+	// and NewSubsystemLoggingHTTPTransport when logging an HTTP request HTTP version via tflog.
+	FieldHttpRequestProtoVersion = "tf_http_req_version"
+
+	// FieldHttpRequestBody is the field key used by NewLoggingHTTPTransport
+	// and NewSubsystemLoggingHTTPTransport when logging an HTTP request body via tflog.
+	FieldHttpRequestBody = "tf_http_req_body"
+
+	// FieldHttpResponseProtoVersion is the field key used by NewLoggingHTTPTransport
+	// and NewSubsystemLoggingHTTPTransport when logging an HTTP response protocol version via tflog.
+	FieldHttpResponseProtoVersion = "tf_http_res_version"
+
+	// FieldHttpResponseStatusCode is the field key used by NewLoggingHTTPTransport
+	// and NewSubsystemLoggingHTTPTransport when logging an HTTP response status code via tflog.
+	FieldHttpResponseStatusCode = "tf_http_res_status_code"
+
+	// FieldHttpResponseStatusReason is the field key used by NewLoggingHTTPTransport
+	// and NewSubsystemLoggingHTTPTransport when logging an HTTP response status reason phrase via tflog.
+	FieldHttpResponseStatusReason = "tf_http_res_status_reason"
+
+	// FieldHttpResponseBody is the field key used by NewLoggingHTTPTransport
+	// and NewSubsystemLoggingHTTPTransport when logging an HTTP response body via tflog.
+	FieldHttpResponseBody = "tf_http_res_body"
+
+	// MessageHttpRequest is the message used by NewLoggingHTTPTransport
+	// and NewSubsystemLoggingHTTPTransport when logging an HTTP request via tflog.
+	MessageHttpRequest = "Sending HTTP Request"
+
+	// MessageHttpResponse is the message used by NewLoggingHTTPTransport
+	// and NewSubsystemLoggingHTTPTransport when logging an HTTP response via tflog.
+	MessageHttpResponse = "Received HTTP Response"
```

Comment:
Should these messages be exported in the public interface? They're okay to be constants, but I'm not sure we want to signal any sort of compatibility or code-desirable use cases for the string values themselves.

Reply:
I was conflicted indeed. Yeah, I'll make them private, good call.